cuda/btx_cudainterval_callbacks.cpp

```diff
@@ -0,0 +1,443 @@
+  }
+}
+
+static void non_kernel_task_stream_present_entry_callback(
+    void *btx_handle, void *usr_data, int64_t ts, const char *event_class_name,
+    const char *hostname, int64_t vpid, uint64_t vtid, CUstream cuStream) {
+  task_entry_helper(btx_handle, usr_data, ts, event_class_name, hostname, vpid,
+                    vtid, cuStream);
+}
+
+static void non_kernel_task_stream_absent_entry_callback(
+    void *btx_handle, void *usr_data, int64_t ts, const char *event_class_name,
+    const char *hostname, int64_t vpid, uint64_t vtid) {
+  task_entry_helper(btx_handle, usr_data, ts, event_class_name, hostname, vpid,
+                    vtid, nullptr);
+}
+
+static void kernel_task_stream_present_entry_callback(
+    void *btx_handle, void *usr_data, int64_t ts, const char *event_class_name,
+    const char *hostname, int64_t vpid, uint64_t vtid, CUstream cuStream,
+    CUfunction f) {
+  task_entry_helper(btx_handle, usr_data, ts, event_class_name, hostname, vpid,
+                    vtid, cuStream, f);
+}
+
+static void kernel_task_stream_absent_entry_callback(
+    void *btx_handle, void *usr_data, int64_t ts, const char *event_class_name,
+    const char *hostname, int64_t vpid, uint64_t vtid, CUfunction f) {
+  task_entry_helper(btx_handle, usr_data, ts, event_class_name, hostname, vpid,
+                    vtid, nullptr, f);
+}
+
+// ===============================
+// END Kernel name and device tracking
+// ===============================
+
+void btx_register_usr_callbacks(void *btx_handle) {
+  btx_register_callbacks_initialize_component(btx_handle,
+                                              &btx_initialize_component);
+  btx_register_callbacks_finalize_component(btx_handle,
+                                            &btx_finalize_component);
+
+  // generic callbacks, for host events
+  btx_register_callbacks_entries(btx_handle, &entries_callback);
+  btx_register_callbacks_exits_cudaError_absent(
+      btx_handle, &exits_callback_cudaError_absent);
+  btx_register_callbacks_exits_cudaError_present(
+      btx_handle, &exits_callback_cudaError_present);
+
+  // Traffic
+  btx_register_callbacks_entries_traffic_v1(btx_handle,
+                                            &entries_traffic_v1_callback);
+  btx_register_callbacks_entries_traffic_v2(btx_handle,
+                                            &entries_traffic_v2_callback);
+  btx_register_callbacks_exits_traffic(btx_handle, &exits_traffic_callback);
+
+  // device profiling events
+  btx_register_callbacks_lttng_ust_cuda_profiling_event_profiling(
+      btx_handle, &profiling_callback);
+  btx_register_callbacks_lttng_ust_cuda_profiling_event_profiling_results(
+      btx_handle, &profiling_callback_results);
+
+  // Context handling
+  btx_register_callbacks_primary_ctx_retain_entry(
+      btx_handle, &primary_ctx_retain_entry_callback);
+  btx_register_callbacks_primary_ctx_retain_exit(
+      btx_handle, &primary_ctx_retain_exit_callback);
```

Comment:
I guess we can create a MACRO too to avoid code bloat, and enforce consistent naming? Not important

Reply:
good idea! was quick to do.